Add docs and minor fixes (#6)
yiisoft/rbac-rules-container: README.md

[![static analysis](https://github.com/yiisoft/rbac-rules-container/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/rbac-rules-container/actions?query=workflow%3A%22static+analysis%22)
[![type-coverage](https://shepherd.dev/github/yiisoft/rbac-rules-container/coverage.svg)](https://shepherd.dev/github/yiisoft/rbac-rules-container)

> This package is a factory for creating [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) rules. It 
> provides rules container wrapping [Yii Factory](https://github.com/yiisoft/factory) and uses 
> [Yii Definitions](https://github.com/yiisoft/definitions) syntax. RBAC manager considers rule as a name and parameters
> and unaware of its creation by design, delegating creation to rules container keeping responsibilities separation.

Reviewer: ```suggestion
This package is a factory for creating [Yii RBAC (Role-Based Access Control)](https://github.com/yiisoft/rbac) rules. It 
provides rules container based on [Yii Factory](https://github.com/yiisoft/factory) and uses 
[Yii Definitions](https://github.com/yiisoft/definitions) syntax. RBAC manager accepts rule as a name and parameters
and is unaware of its creation by design, delegating creation to rules container keeping responsibilities separation.
```

Author: Fun fact about "based on" - that's how it was initially. :slightly_smiling_face:

## General usage

### Direct interaction with rules container

```php
use Psr\Container\ContainerInterface;
use Yiisoft\Rbac\Item;
use Yiisoft\Rbac\RuleInterface;
use Yiisoft\Rbac\Rules\Container\RulesContainer;

/**
 * Checks if user ID matches `authorID` passed via parameters.
 */
final class AuthorRule implements RuleInterface
{
    public function execute(string $userId, Item $item, array $parameters = []): bool
    {
        return $parameters['authorID'] === $userId;
    }
}

$rulesContainer = new RulesContainer(new MyContainer());
$rule = $rulesContainer->create(AuthorRule::class);
```

- `MyContainer` is a container for resolving dependencies and  must be an instance of 

> `Psr\Container\ContainerInterface`.  [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 

Reviewer 2: ```suggestion
`Psr\Container\ContainerInterface`. [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 
```

> - `MyContainer` is a container for resolving dependencies and  must be an instance of 
> `Psr\Container\ContainerInterface`.  [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 

Reviewer: ```suggestion
- `MyContainer` is a container for resolving dependencies and must be an instance of 
`Psr\Container\ContainerInterface`. [Yii Dependency Injection](https://github.com/yiisoft/di) implementation also can 
```

Author: Fixed in https://github.com/yiisoft/rbac-rules-container/pull/6#discussion_r845453019.

be used.
- You can optionally set [definitions](https://github.com/yiisoft/definitions) and disable their validation if needed.

Basically, the arguments are the same as in [Yii Factory](https://github.com/yiisoft/factory). Please refer to its docs 
for more details.

Rules are created only once, then cached and reused for repeated calls.

```php
$rule = $rulesContainer->create(AuthorRule::class); // Returned from cache
````

> ### Using config

Reviewer 2: Need info that this config for config plugin.
